DQMOffline/Trigger/plugins/HTMonitor.cc

```diff
@@ -0,0 +1,301 @@
+  bookME(ibooker,htVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,ht_binning_.xmin, ht_binning_.xmax);
+  setHTitle(htVsLS_,"LS","HT [GeV]");
+
+  histname = "deltaphi_metjet1"; histtitle = "DPHI_METJ1";
+  bookME(ibooker,deltaphimetj1ME_,histname,histtitle,phi_binning_.nbins, phi_binning_.xmin, phi_binning_.xmax);
+  setHTitle(deltaphimetj1ME_,"delta phi (met, j1)","events / 0.1 rad");
+
+  histname = "deltaphi_jet1jet2"; histtitle = "DPHI_J1J2";
+  bookME(ibooker,deltaphij1j2ME_,histname,histtitle,phi_binning_.nbins, phi_binning_.xmin, phi_binning_.xmax);
+  setHTitle(deltaphij1j2ME_,"delta phi (j1, j2)","events / 0.1 rad");
+
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void HTMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+
+  float ht = 0.0; 
+
+  edm::Handle<reco::PFJetCollection> jetHandle; //add a configurable jet collection & jet pt selection
+  iEvent.getByToken( jetToken_, jetHandle );
+  std::vector<reco::PFJet> jets;
+  jets.clear();
+  if ( jetHandle->size() < njets_ ) return;
+  for ( auto const & j : *jetHandle ) {
+    if ( jetSelection_( j ) ) {
+      jets.push_back(j);
+      if ( jetSelection_HT_(j)) ht += j.pt();
+    }
+  }
+  for ( auto const & j : *jetHandle ) {
+    if ( jetSelection_HT_(j)) ht += j.pt();
+  }
+  if ( jets.size() < njets_ ) return;
+
+  float deltaPhi_met_j1 = 10.0;
+  float deltaPhi_j1_j2 = 10.0;
+
+  if (jets.size() >= 1) deltaPhi_met_j1 = fabs( deltaPhi( pfmet.phi(),  jets[0].phi() ));
+  if (jets.size() >= 2) deltaPhi_j1_j2 = fabs( deltaPhi( jets[0].phi(),  jets[1].phi() ));
+  
+  edm::Handle<reco::GsfElectronCollection> eleHandle;
+  iEvent.getByToken( eleToken_, eleHandle );
+  std::vector<reco::GsfElectron> electrons;
+  electrons.clear();
+  if ( eleHandle->size() < nelectrons_ ) return;
+  for ( auto const & e : *eleHandle ) {
+    if ( eleSelection_( e ) ) electrons.push_back(e);
+  }
+  if ( electrons.size() < nelectrons_ ) return;
+  
+  edm::Handle<reco::VertexCollection> vtxHandle;
+  iEvent.getByToken(vtxToken_, vtxHandle);
+
+  reco::Vertex vtx;
+  math::XYZPoint pv(0, 0, 0);
```

Comment:
This code was already present before I began working on this. If you would like me to make use of beam spot and if  @rappoccio agrees that this is a necessary update then I will implement it.